### PR TITLE
3454 use earthly for building webui

### PR DESCRIPTION
**Pull request by aronchick**
## Summary by CodeRabbit

- **New Features**
	- Introduced Earthly for improved build processes.
	- Expanded CircleCI workflow triggers to include more pull request actions and pushes to the main branch.
	- Added new commands and parameters to enhance build configuration.

- **Bug Fixes**
	- Removed redundant yarn commands to streamline build jobs.

- **Documentation**
	- Updated the structure of `webui/index.html` for better readability and maintenance.

- **Refactor**
	- Refactored build commands and configurations for efficiency.
	- Updated alias paths in `vite.config.mjs` for clearer project structure.

- **Chores**
	- Updated Docker image version and authentication details.
	- Added `earthly 0.8.3` to the tool versions for consistent build environments.
	- Removed unnecessary configuration settings in `.yarnrc.yml`.

- **Tests**
	- Defined steps for running linter checks and tests in the new Earthfile.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,7 +56,10 @@ executors:
     resource_class: windows.xlarge
   docker-linux:
     docker:
-      - image: cimg/base:2022.01
+      - image: cimg/base:2022.09
+        auth:
+          username: $DOCKERHUB_USERNAME
+          password: $DOCKERHUB_PASSWORD
```
[the remark this answers is not in the document]
Comment (aronchick) [reply .circleci/config.yml:62]: Uses env variables

File: .circleci/config.yml [hunk 2]
```diff
@@ -238,6 +251,16 @@
...
             popd
+  install_earthly:
+    description: Install Earthly and dependencies.
+    steps:
+      - run:
+          name: Install Earthly
+          command: |
+            sudo /bin/sh -c "wget 'https://github.com/earthly/earthly/releases/latest/download/earthly-linux-amd64' -O /usr/local/bin/earthly"
+            sudo chmod +x /usr/local/bin/earthly
+            /usr/local/bin/earthly bootstrap
```
[the remark this answers is not in the document]
Comment (aronchick) [reply .circleci/config.yml:262]: replacing with asdf

[the remark this answers is not in the document]
Comment (aronchick) [reply .circleci/config.yml:262]: we already use asdf

File: webui/index.html [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-    <title>Bacalhau WebUI</title>
-  </head>
+  <link rel="manifest" href="/images/manifest.json" />
+  <title>Bacalhau WebUI</title>
+</head>
+
+<body>
+  <noscript>You need to enable JavaScript to run this app.</noscript>
+  <div id="root"></div>
+  <script type="module" src="/src/index.tsx"></script>
```
[the remark this answers is not in the document]
Comment (aronchick) [reply webui/index.html:25]: these are in the body

**Comment by rossjones** [comment]
I tried to run this locally with `make clean && make build` and it complained

```
make: *** No rule to make target `webui/build/index.html', needed by `bin/darwin/arm64/bacalhau'.  Stop.
```

File: Makefile [hunk 4]
```diff
@@ -163,31 +165,11 @@ build-dev: build-ci
...
-$(WEB_INSTALL_GUARD): webui/node_modules webui/package.json
-	@echo "Updating webui dependencies"
-	@cd webui && yarn install
-	touch $(WEB_INSTALL_GUARD)
-
-export GENERATE_SOURCEMAP := false
-${WEB_BUILD_GUARD}: $(WEB_INSTALL_GUARD) $(WEB_SRC_FILES)
-	cd webui && yarn run build
-	touch $(WEB_BUILD_GUARD)
+build-webui: $(WEB_INSTALL_GUARD) $(WEB_SRC_FILES)
```
Comment (rossjones) [inline Makefile:170]: $(WEB_INSTALL_GUARD) doesn't exist any more.

File: webui/vite.config.mjs [hunk 5]
```diff
@@ -28,7 +28,9 @@ export default defineConfig(({ mode }) => {
     target: browserslistToEsbuild([">0.2%", "not dead", "not op_mini all"]),
     resolve: {
       alias: {
-        "@": fileURLToPath(new URL("./src", import.meta.url)),
+        // "@": fileURLToPath(new URL("./src", import.meta.url)),
```
Comment (rossjones) [inline webui/vite.config.mjs:31]:
```suggestion
```

File: Makefile (hunk 4 above)
Comment (aronchick) [reply Makefile:170]: thanks

**Comment by aronchick** [comment]
> I tried to run this locally with `make clean && make build` and it complained
> 
> ```
> make: *** No rule to make target `webui/build/index.html', needed by `bin/darwin/arm64/bacalhau'.  Stop.
> ```

fixed!

**Review by rossjones — APPROVED** [review]
👍🏻

**Pull request merged**